### PR TITLE
[TS migration] Migrate RoomInvitePage to typescript

### PR DESCRIPTION
**Pull request by SzymczakJ**
### Fixed Issues

$ https://github.com/Expensify/App/issues/31993
PROPOSAL: N/A

### Tests

- [x] Test main flows of the app(sending messages, logging, requesting money)
- [x] Create room and try to invite users, verify that everything works on invite page.

### Offline tests

N/A
### QA Steps

Same as tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/88395093/244cc268-d035-42c8-a24a-bce08765f5cf

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/88395093/633652ec-c155-4434-9c39-d010c129b1ab

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/88395093/6791feac-2d7e-4189-ac53-0d03abd377b4

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/88395093/29308e65-e385-4e4e-ae07-4e4259a7c632

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/88395093/f209203e-37d9-4516-93f3-d1f5c7607daf

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/88395093/ccf5b1be-a637-4dbd-9b9a-0c191645a328

</details>

### CODE REVIEW

File: src/pages/RoomInvitePage.tsx [hunk 1]
```diff
@@ -184,60 +160,51 @@ function RoomInvitePage(props) {
...
     return (
         <ScreenWrapper
             shouldEnableMaxHeight
             testID={RoomInvitePage.displayName}
         >
             <FullPageNotFoundView
-                shouldShow={_.isEmpty(props.report)}
-                subtitleKey={_.isEmpty(props.report) ? undefined : 'roomMembersPage.notAuthorized'}
+                shouldShow={!report}
+                subtitleKey={!report ? undefined : 'roomMembersPage.notAuthorized'}
```
Comment (blazejkustra) [inline src/pages/RoomInvitePage.tsx:207]: For objects we need to use `isEmptyObject` util instead:
```suggestion
                shouldShow={isEmptyObject(report)}
                subtitleKey={isEmptyObject(report)? undefined : 'roomMembersPage.notAuthorized'}
```
```suggestion
                shouldShow={!report}
                subtitleKey={!report ? undefined : 'roomMembersPage.notAuthorized'}
```

**Review by blazejkustra — CHANGES_REQUESTED** [review]

**Review by blazejkustra — APPROVED** [review]

File: src/libs/OptionsListUtils.ts [hunk 2]
```diff
@@ -1833,14 +1828,13 @@ function formatMemberForList(member: ReportUtils.OptionData, config: ReportUtils
         alternateText: member.alternateText || member.login || '',
         // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
         keyForList: member.keyForList || String(accountID ?? 0) || '',
-        isSelected: false,
-        isDisabled: false,
+        isSelected: member.isSelected ? member.isSelected : false,
```
Comment (kubabutkiewicz) [inline src/libs/OptionsListUtils.ts:1831]:
```suggestion
        isSelected:  member.isSelected ?? false,
```
will that work ?

File: src/libs/OptionsListUtils.ts [hunk 3]
```diff
@@ -1833,14 +1828,13 @@ function formatMemberForList(member: ReportUtils.OptionData, config: ReportUtils
...
+        isDisabled: member.isDisabled ? member.isDisabled : false,
```
Comment (kubabutkiewicz) [inline src/libs/OptionsListUtils.ts:1832]: same here

File: src/pages/RoomInvitePage.tsx [hunk 4]
```diff
@@ -121,57 +105,55 @@ function RoomInvitePage(props) {
...
-        const hasUnselectedUserToInvite = userToInvite && !_.contains(selectedLogins, userToInvite.login);
+        const selectedLogins = selectedOptions.map(({login}) => login);
+        const personalDetailsWithoutSelected = invitePersonalDetails.filter(({login}) => !selectedLogins.includes(login));
+        const personalDetailsFormatted = personalDetailsWithoutSelected.map((personalDetail) => OptionsListUtils.formatMemberForList(personalDetail));
+        const hasUnselectedUserToInvite = userToInvite && !selectedLogins.includes(userToInvite.login);
 
         sectionsArr.push({
             title: translate('common.contacts'),
             data: personalDetailsFormatted,
-            shouldShow: !_.isEmpty(personalDetailsFormatted),
```
Comment (kubabutkiewicz) [inline src/pages/RoomInvitePage.tsx:151]: why this was removed?

Comment (SzymczakJ) [reply src/pages/RoomInvitePage.tsx:151]: shouldShow is not used anywhere in SelectionList(sectionArr is data for SectionList) so I've removed it.
Double checked if everything works.

**Comment by SzymczakJ** [comment]
@kubabutkiewicz fixed comments 😅

**Review by kubabutkiewicz — APPROVED** [review]

**Comment by SzymczakJ** [comment]
@kubabutkiewicz fixed comments

**Comment by mananjadhav** [comment]
I’ll check this once I am online. Quick question, is the unit test failure related to this PR?

**Comment by SzymczakJ** [comment]
@mananjadhav tests failure was related to this PR but I've already fixed them

File: src/libs/OptionsListUtils.ts [hunk 5]
```diff
@@ -1837,12 +1836,8 @@ function getShareDestinationOptions(
  * @param member - personalDetails or userToInvite
  * @param config - keys to overwrite the default values
  */
-function formatMemberForList(member: ReportUtils.OptionData, config: ReportUtils.OptionData | EmptyObject = {}): MemberForList | undefined {
```
Comment (mananjadhav) [inline src/libs/OptionsListUtils.ts:1840]: have we checked all usages to see that it's okay to remove `config`?

File: src/libs/OptionsListUtils.ts [hunk 6]
```diff
@@ -1837,12 +1836,8 @@ function getShareDestinationOptions(
...
-    if (!member) {
```
Comment (mananjadhav) [inline src/libs/OptionsListUtils.ts:1841]: do we have any case where this could error?

Comment (SzymczakJ) [reply src/libs/OptionsListUtils.ts:1841]: Same as above, every usage of this function has member of ReportUtils.OptionData type, not undefined or null, so we don't need this check. Future usages are going to be checked by typecheck.

File: src/libs/OptionsListUtils.ts (hunk 5 above)
Comment (SzymczakJ) [reply src/libs/OptionsListUtils.ts:1840]: Double checked if everything works. This method is used only in two places and none of them use `config` param.

File: src/pages/RoomInvitePage.tsx [hunk 7]
```diff
@@ -26,77 +27,60 @@ import * as Report from '@userActions/Report';
...
+
+function RoomInvitePage({betas, personalDetails, report, policies}: RoomInvitePageProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const [searchTerm, setSearchTerm] = useState('');
-    const [selectedOptions, setSelectedOptions] = useState([]);
-    const [personalDetails, setPersonalDetails] = useState([]);
-    const [userToInvite, setUserToInvite] = useState(null);
+    const [selectedOptions, setSelectedOptions] = useState<ReportUtils.OptionData[]>([]);
+    const [invitePersonalDetails, setPersonalDetails] = useState<ReportUtils.OptionData[]>([]);
```
Comment (mananjadhav) [inline src/pages/RoomInvitePage.tsx:49]: should we rename the set state method? I was a bit confused when tracking where was `invitePersonalDetails` updated.

```suggestion
    const [invitePersonalDetails, setInvitePersonalDetails] = useState<ReportUtils.OptionData[]>([]);
```

File: src/pages/RoomInvitePage.tsx [hunk 8]
```diff
@@ -121,34 +105,33 @@ function RoomInvitePage(props) {
         // Filter all options that is a part of the search term or in the personal details
         let filterSelectedOptions = selectedOptions;
         if (searchTerm !== '') {
-            filterSelectedOptions = _.filter(selectedOptions, (option) => {
-                const accountID = lodashGet(option, 'accountID', null);
-                const isOptionInPersonalDetails = _.some(personalDetails, (personalDetail) => personalDetail.accountID === accountID);
+            filterSelectedOptions = selectedOptions.filter((option) => {
+                const accountID = option?.accountID;
+                const isOptionInPersonalDetails = invitePersonalDetails.some((personalDetail) => accountID && personalDetail.accountID && personalDetail.accountID === accountID);
```
Comment (mananjadhav) [inline src/pages/RoomInvitePage.tsx:110]:
```suggestion
                const isOptionInPersonalDetails = invitePersonalDetails.some((personalDetail) => accountID && personalDetail?.accountID === accountID);
```

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/3069065/b16b6576-a7ec-4932-86fc-3a956b6d5dae

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/3069065/2655f9ef-843a-42ab-b048-d98fc419d97c

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/3069065/c4a3e0a9-9c0b-4a36-babd-c00205503b0b

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/3069065/cf398d4a-d6fa-4b6b-b003-180c639e541a

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/3069065/a9fd0e74-9ace-4845-ae0d-6ea7644175b4

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/3069065/776a28b2-b160-4f7f-a2cc-4873eba9ab8d

</details>

File: src/pages/RoomInvitePage.tsx (hunk 7 above)
Comment (SzymczakJ) [reply src/pages/RoomInvitePage.tsx]: Good point, should have renamed it when I was renaming personalDetails into invitePersonalDetails

**Comment by SzymczakJ** [comment]
@mananjadhav fixed comments

**Review by mananjadhav — APPROVED** [review]

**Review by mananjadhav — APPROVED** [review]

**Review by aldo-expensify — APPROVED** [review]

**Pull request merged**